src/Admin/AbstractAdmin.php

```diff
@@ -2391,8 +2528,35 @@ public function getFilterTheme()
         return $this->filterTheme;
     }
 
-    public function addExtension(AdminExtensionInterface $extension)
+    public function addExtension(OldAdminExtensionInterface $extension)
+    {
+        @trigger_error(
+            'Method "addExtension" is deprecated. Check "addLightweightExtension".',
+            E_USER_DEPRECATED
+        );
+
+        $this->extensions[] = $extension;
+    }
+
+    /**
+     * NEXT_MAJOR: remove whole if and add only NewAdminExtensionInterface typehint
+     *
+     * @param NewAdminExtensionInterface|OldAdminExtensionInterface $extension
+     */
+    public function addLightweightExtension($extension)
```

Comment:
So this is the new method, don't really like the name :/

Reply:
`addListener`, `addSubscriber`, `addHook`, maybe?